types/src/legacy_sse_data/fixtures.rs

```diff
@@ -77,6 +80,26 @@ pub fn legacy_deploy_processed() -> LegacySseData {
     serde_json::from_str(RAW_LEGACY_DEPLOY_PROCESSED).unwrap()
 }
 
+pub fn block_v2() -> BlockV2 {
+    serde_json::from_str(BLOCK_BODY_V2_WITH_ALL_DATA).unwrap()
+}
+
+pub fn block_v1_no_deploys_no_era() -> BlockV1 {
+    let mut val = serde_json::from_str::<Value>(BLOCK_BODY_V1_ALL_DATA).unwrap();
```

Comment:
This is done through json deserialization because BlockBody* structures from casper-types don't offer `new` functions and/or setters of their fields

Reply:
have you tried using `TestBlockV1Builder`? I think it might simplify this

Reply:
:+1: